tests/cases/compiler/uncalledFunctionChecksInConditionalPerf.ts

```diff
@@ -0,0 +1,5 @@
+// @strictNullChecks: true
```

Comment:
I wonder if we have a better way of capturing this kind of test, where the thing we're interested in is the execution time.

Reply:
Historically we don't - but tests do have a timeout (around 50s), so if you make sure they're large enough to hit the timeout before your change, they're a decent test; they won't capture tiny pretty regressions, but will prevent big ones. We have other tests in this vein.

Reply:
Ok then. @JoostK can you make this example much much bigger, so that the test would timeout without the change?

Reply:
I will add 4 additional binary ops, which would have brought check times up by a factor of 16 (easily pushing it beyond the 50s threshold)